lib/syntax_tools/src/erl_syntax.erl

```diff
@@ -930,8 +930,12 @@ set_pos(Node, Pos) ->
     case Node of
 	#tree{attr = Attr} ->
 	    Node#tree{attr = Attr#attr{pos = Pos}};
-	#wrapper{attr = Attr} ->
-	    Node#wrapper{attr = Attr#attr{pos = Pos}};
+	#wrapper{attr = Attr, tree = {error, {_, Module, Reason}}} ->
+	    Node#wrapper{attr = Attr#attr{pos = Pos}, tree = {error, {Pos, Module, Reason}}};
+    #wrapper{attr = Attr, tree = {warning, {_, Module, Reason}}} ->
+        Node#wrapper{attr = Attr#attr{pos = Pos}, tree = {warning, {Pos, Module, Reason}}};
+	#wrapper{attr = Attr, tree = Tree} ->
+	    Node#wrapper{attr = Attr#attr{pos = Pos}, tree = setelement(2, Tree, Pos)};
```

Comment:
Nitpick: the indentation is a bit off. While you're at it, you can convert tabs to 8 spaces here.

Reply:
I think I got rid of all the tabs now.